Modules/+Drivers/+Kinesis/KinesisBSC203.m

```diff
@@ -0,0 +1,347 @@
+                try
+                    obj.deviceNET.Disconnect(true)
+                catch
+                    error(['Unable to disconnect device',obj.serialnumbers{obj.motor_channels(i)}]);
+                end
+                obj.isconnected = obj.deviceNET.IsConnected();
+            else % Cannot disconnect because device not connected
+                error('Device not connected.')
+            end
+        end
+
+        function home(obj)
+            for i = 1:3
+                if ~isnan(obj.motor_channels(i))
+                    workDone=obj.channelsNET{obj.motor_channels(i)}.InitializeWaitHandler();     % Initialise Waithandler for timeout
+                    obj.channelsNET{obj.motor_channels(i)}.Home(workDone);                       % Home device via .NET interface
+                    obj.channelsNET{obj.motor_channels(i)}.Wait(obj.TIMEOUTMOVE);                % Wait for move to finish     
+                end                 
+            end
+            obj.updatestatus; % Update status variables from device
+        end
+
+        function tf = checkMove(obj, target_pos)
+            %   Check to make sure target_pos is ok to execute
+            %   Error if it is outside limits
+            %   Error if the channel needs to be homed
+            %   Otherwise returns true
+            tf = true;
+            for i = 1:3
+                if ~isnan(obj.motor_channels(i))
+                    assert(~obj.channelsNET{obj.motor_channels(i)}.NeedsHoming,'Motor %f is not homed!', i)
+                    assert(target_pos(i) <= max(obj.Travel) && target_pos(i) >= min(obj.Travel),...
+                        'Attempted to move motor %f to %f, but it is limited to %f, %f', obj.motor_channels(i), target_pos, min(obj.Travel), max(obj.Travel))
+                end
+            end
+        end
+
+        function moveto(obj, target_pos)
+            %   Move to target position, target_pos := 1 * 3 array of double
+            tf = obj.checkMove(target_pos);
+            if tf
+                target_pos_calib = target_pos / obj.calibration;
+                obj.isMoving = true;
+                for i = 1:3
+                    if ~isnan(obj.motor_channels(i))
+                        try
+                            workDone=obj.channelsNET{obj.motor_channels(i)}.InitializeWaitHandler(); % Initialise Waithandler for timeout
+                            obj.channelsNET{obj.motor_channels(i)}.MoveTo(target_pos_calib(i), workDone);       % Move device to position via .NET interface
+                            obj.channelsNET{obj.motor_channels(i)}.Wait(obj.TIMEOUTMOVE);              % Wait for move to finish
+                        catch
+                            error(['Unable to Move channel ',obj.serialnumbers{obj.motor_channels(i)},' to ',num2str(target_pos(i))]);
+                        end
+                    end
+                end
+            else
+                error('Target position is out of range')
```

Comment:
Could shorten this with `assert(obj.checkMove(target_pos),'Target position is out of range')`